util/testutil/integration/containerd.go

```diff
@@ -160,7 +161,17 @@ disabled_plugins = ["cri"]
 	}
 	deferF.append(stop)
 
-	return &cdsandbox{address: address, sandbox: sandbox{mv: conf.mv, address: buildkitdSock, logs: logs, cleanup: deferF, rootless: false}}, cl, nil
+	return &cdsandbox{
```

Comment:
This should return a clean `backend` or `cdbackend` without all the embedded sandboxes. Ideally, a backend implementation can be defined outside of this package and shouldn't require anything private in here. I see you need access to some things like logs, conf and user so I guess it should be smth like 

```
type BackendOpt struct {
  Logs map[]..
  UID, GID int
  Config string
}
New(BackendOpt) (Backend, func() error, error)
```

Reply:
The problem here is that the logs and what not end up being part of the sandbox, so if we have them in new I don't see a clean way to get them into the sandbox too. I suppose I could have the built in backend implementations implement `Sandbox` too, and then when you do `newSandbox` if it's already a sandbox it can just return it. This way external implementations could do the same thing if they want to have their own logs already done, or they could just implement the `Backend` methods. This doesn't feel great though.

Reply:
Ah, I see, I still had the flow backwards. I think the sandbox abstraction still doesn't make sense to me, but I'll tweak it to work this way and see how it feels.

Reply:
@tonistiigi I think I've got something that's a bit more straight forward now. PTAL and let me know if it's closer to what you had in mind.